### DISCUSSION TITLE
Web Service Management - Removing a Web Service?

### DISCUSSION THREAD

**Discussion by bjarkihall** [microsoft/ALAppExtensions#14667] [Q&A]
Hi, AppSourceCop rule [AS0059](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/appsourcecop-as0059) prevents Insert/Modify/Delete for `Record "Web Service Aggregate"`.

We can use `CreateTenantWebService` to Insert/Modify:
https://github.com/microsoft/ALAppExtensions/tree/main/Modules/System/Web%20Service%20Management#createtenantwebservice-method-

We can modify the service with `Published = false`, but there's no way to "clean-up" declared web services (which Delete used to be able to do), which can leave a lot of unused web services in the `Web Services` list.

Would MS be open to a Pull-Request which exposes something like `RemoveTenantWebService`?

**Comment by bjarkihall** [comment]
@JesperSchulz sorry for tagging you like this, but am I posting this kind of question in the wrong place (also https://github.com/microsoft/ALAppExtensions/discussions/14325)?

I thought these would make sense to "discuss" before diving fully into forking, implementing, writing tests, creating pull request, someone at MS gets assigned for reviewing the code changes, when there might be a good reason for these not to be available in the SystemApp in the first place. :)

Are "issues" the preferred way of getting initial feedback if something would even be considered for a PR?

**Reply by SimonOfHH** [reply]
Hi @bjarkihall 
good question (regarding getting feedback). I think Github Discussions is still a somewhat new form of feedback for the BC-team. But I think in your specific case it might make more sense to open an issue. I'd think of the Discussions-section as used for new ideas/features. But the thing you are mentioning really sounds more as a regular issue.

Not sure if I'm right, though.

**Reply by JesperSchulz** [reply]
The Discussion section doesn't get that much attention yet; I'll try to change that! It is meant as a forum for the community to discuss extensibility issues, PR suggestions and the likes of it. Hence your post as such was valid enough! However, the forum is not supposed to be yet another Microsoft support platform, where developers ask and Microsoft answers - we've got plenty of those already ;-) I am hoping that over time, the entire community will discuss everything extensibility related in that forum, and possibly create linked PRs based on discussions. I guess for now, creating an issue is the better way, if you want to point out a specific problem with extensibility specifically to us.
Seems like you did and seems like you got an answer?
